Assets/Mirror/Components/NetworkTransformBase.cs

```diff
@@ -221,16 +221,18 @@ void DeserializeFromReader(NetworkReader reader)
             //
             else
             {
-                float oldDistance = Vector3.Distance(start.localPosition, goal.localPosition);
-                float newDistance = Vector3.Distance(goal.localPosition, temp.localPosition);
+                // SqrMagnitude is faster than Distance per Unity docs
+                // https://docs.unity3d.com/ScriptReference/Vector3-sqrMagnitude.html
+                float oldDistance = Vector3.SqrMagnitude(start.localPosition - goal.localPosition);
+                float newDistance = Vector3.SqrMagnitude(goal.localPosition - temp.localPosition);
 
                 start = goal;
 
                 // teleport / lag / obstacle detection: only continue at current
                 // position if we aren't too far away
-                //
-                // // local position/rotation for VR support
-                if (Vector3.Distance(targetComponent.transform.localPosition, start.localPosition) < oldDistance + newDistance)
+                // local position/rotation for VR support
+                // oldDistance and newDistance are already squared above...don't square them again here
+                if (Vector3.SqrMagnitude(targetComponent.transform.localPosition - start.localPosition) < oldDistance + newDistance)
```

Comment:
This code has a bug in it. The sum of squares is not equal to the square of sums.
Original code simplified: `if (c < a + b)`
Proposed code simplified: `if (c^2 < a^2 + b^2)`
This is wrong, if you square both sides it would be `if (c^2 < (a + b)^2)`
which is equal to `if (c^2 < a^2 + 2ab b^2)`
Meaning you need to square root both a^2 and b^2 to calculate the 2ab term.
PR is well intentioned though. I can help fix it, but it would be more complicated (but still faster).

Reply:
Feel free to; ping me when done and I'll do more benchmarking with the fix.

Reply:
this is exactly what we tried to avoid.
everyone thinks micro optimizations are easy and obvious. but they aren't, as evident by the fact that 10 people looked at it and missed the bug until @rodolphito saw it.
I am closing this again. this is crazy.

Reply:
So you're saying you don't care about mobile performance optimizations, vis?

Reply:
Everyone looked at this code and none of us saw this bug for 3 days.
Do you guys still think "this is free"?

This is exactly why I didn't want to merge this,  it goes from obvious code to complicated and evidently error prone code.

Reply:
I didn't look at it, I merely pointed out that sqrMag is always faster. But I'll add, 'someone made a math mistake lets not implement this properly' to my list of dumb reasons why this PR is closed. I guess with that kind of thinking you should just delete any new feature or improvement that doesn't work immediately.

Reply:
The fix only needs one sqrt:
`if (c^2 < a^2 + 2 * sqrt(a^2 * b^2) + b^2)`
Reasoning: `sqrt(a * b) = sqrt(a) * sqrt(b)` so we don't need to sqrt a^2 and b^2 separately. 
It can also be slightly sped up by only computing the sqrt if a preliminary check passes:
`if (c^2 < a^2 + b^2 || c^2 < a^2 + sqrt(4 * a^2 * b^2) + b^2)`
Reasoning: we are working with distances so all terms are positive. if `c^2 < a^2 + b^2`, adding a positive number to the right hand side will not affect that. If the `c^2 < a^2 + b^2` check fails, we do the full check with the `sqrt(4 * a^2 * b^2)` term to be sure.